### PR TITLE
Add locale switcher in the page chooser

### PR DESCRIPTION
**Pull request by zerolab**
This PR adds the locale switcher in the page chooser.

Screencast for the functionality: https://www.loom.com/share/1a0bbbfed4a04544a2575917190a66e0

### CODE REVIEW

File: wagtail/admin/views/chooser.py [hunk 1]
```diff
@@ -129,10 +130,54 @@ def browse(request, parent_page_id=None):
     parent_page.can_choose = can_choose_page(
         parent_page, permission_proxy, desired_classes, can_choose_root, user_perm, target_pages=target_pages, match_subclass=match_subclass)
 
+    selected_locale = None
+    locale_options = []
     show_locale_labels = getattr(settings, 'WAGTAIL_I18N_ENABLED', False)
     if show_locale_labels:
         pages = pages.select_related('locale')
 
+        # 'locale' is the current value of the "Locale" selector in the UI
```
Comment (kaedroho) [inline wagtail/admin/views/chooser.py:139]: I think this part is only relevant when browsing the root page (same with the `if selected_locale` block below)

File: wagtail/admin/views/chooser.py [hunk 2]
```diff
@@ -129,10 +130,54 @@ def browse(request, parent_page_id=None):
...
+                    'locale': locale,
+                    'url': choose_url + '?' + urlencode({
+                        'page_type': page_type_string,
+                        'locale': locale.language_code
+                    })
+                }
+                for locale in Locale.objects.filter(pk__in=pages.values_list('locale_id')).exclude(pk=active_locale_id)
+            ]
+        else:
+            # we have a parent page (that is not the root page), so get the locales based on the available translations
```
Comment (kaedroho) [inline wagtail/admin/views/chooser.py:160]: If we have a parent page, should we set ``selected_locale`` so the locale of the current parent is shown in the dropdown?

File: wagtail/admin/views/chooser.py (hunk 1 above)
Comment (zerolab) [reply wagtail/admin/views/chooser.py]: It is indeed. Tweaked based on this and the comment below.

aside: adding the switcher at the root level was a last minute tweak, so looking at the whole thing with fresh eyes was useful

File: wagtail/admin/views/chooser.py (hunk 2 above)
Comment (zerolab) [reply wagtail/admin/views/chooser.py]: sorted in b8c35c5

**Review by kaedroho — APPROVED** [review]
This looks good to me, thanks!

**Pull request merged**